transformer/metadata.go

```diff
@@ -69,6 +105,30 @@ func (meta *Metadata) Transform(c *skogul.Container) error {
 	return nil
 }
 
+// splitMetricsByObjectKey splits the metrics into multiple metrics based on a key in a list of sub-metrics
+func splitMetricsByObjectKey(metrics *[]*skogul.Metric, metadata *Metadata) ([]*skogul.Metric, error) {
+	origMetrics := *metrics
+	var newMetrics []*skogul.Metric
+
+	// @ToDo: Parse this to a path if more than one element deep (i.e. my.nested.json.structure should unfurl)
+	splitPath := metadata.Split[0]
+
+	for mi := range origMetrics {
+		metricObj := origMetrics[mi].Data[splitPath]
+
```

Comment:
We really should do an explicit cast here, in case we're mistaken about the input data. Right now, it's implicit in the next range-loop, and doesn't check if it succeeds or not... Since this is based on input-data, we can't make too many assumptions.

Reply:
Fixed